docs/spec/staking/end_block.md

```diff
@@ -16,3 +16,34 @@ EndBlock() ValidatorSetChanges
     ClearTendermintUpdates()
     return vsc
 ```
+
+
+## CompleteUnbonding
+
+Complete the unbonding and transfer the coins to the delegate. Perform any
+slashing that occurred during the unbonding period.
+
+```golang
+unbondingQueue(currTime time.Time):
+    for all unbondings whose CompleteTime < currTime:
+        validator = GetValidator(unbonding.ValidatorAddr)
+        returnTokens = ExpectedTokens * unbonding.startSlashRatio/validator.SlashRatio
+        AddCoins(unbonding.DelegatorAddr, returnTokens)
+        removeUnbondingDelegation(unbonding)
+    return
+```
+
+
+
+## CompleteRedelegation
+
+Note that unlike CompleteUnbonding slashing of redelegating shares does not
+take place during completion. Slashing on redelegated shares takes place
+actively as a slashing occurs.
```

Comment:
Then explain why we need this, e.g. "The redelegation completion queue serves simply to clean up state, as redelegations older than an unbonding period need not be kept."